### ISSUE TITLE
[Request] Add twitch.tv support

### ISSUE THREAD

**Issue by gergesh** [TeamNewPipe/NewPipeExtractor#76]
With the direction that the app and extractor are going in, I think this will be a welcome addition.

For the unaware: [Twire](https://f-droid.org/en/packages/com.perflyst.twire/) is a FOSS Twitch app.

**Comment by theScrabi** [comment]
Nnnnnice :D 

going to take a look at it soon :)

**Comment by theScrabi** [comment]
Yes of course. :D 

Help is always welcome, however I'm trying to add twitch support when i work on the extractor [documentation](https://teamnewpipe.github.io/tutorial/)

**Comment by moshpirit** [comment]
The documentation link is no longer available

**Comment by theScrabi** [comment]
https://teamnewpipe.github.io/documentation/

**Comment by opusforlife2** [comment]
Indeed! Here is the news: https://github.com/twireapp/Twire/issues/314#issuecomment-1047837454

**Comment by AudricV** [comment]
Yes, but there is the internal GraphQL API which works without login at all for the features the extractor provides and should be the one the extractor should use.

**Comment by opusforlife2** [comment]
@AudricV See the "allowed" section and Source no. 2 of OP: https://github.com/twireapp/Twire/pull/238

**Comment by Eiion** [comment]
> Indeed! Here is the news: [twireapp/Twire#314 (comment)](https://github.com/twireapp/Twire/issues/314#issuecomment-1047837454)

This thread is about adding twitch.tv support to new pipe. You're just linking to a completely different repository - that's not answering my question.

**Comment by opusforlife2** [comment]
That completely different repository contains the answer to your question.

**Comment by Eiion** [comment]
No, it does not. It's a completely different repository that has nothing to do with New Pipe. And especially when Twire is a thing that doesn't answer my question regarding the progress of adding twitch support to New Pipe when that was requested over for years ago already.
BTW, it would have taken you less effort to give a proper reply than what you've said so far.

**Comment by AudricV** [comment]
> @AudricV See the "allowed" section and Source no. 2 of OP: https://github.com/twireapp/Twire/pull/238

@opusforlife2 Since we won't be allowing logging in to Twitch, using the internal GraphQL API should pose no problem for us; it's no different from our usage of the internal YouTube API. Twire cares about this probably because they allow login (and it is now required because of Helix version of the Twitch official API), and if Twitch detects an account using the internal API from a third-party client, they may ban that account.

@Eiion We still haven't taken a decision about whether to add the Twitch service into the extractor.

Also note that it seems that this internal API is also used by yt-dlp to download Twitch videos.

**Comment by Eiion** [comment]
> @Eiion We still haven't taken a decision about whether to add the Twitch service into the extractor.
> 
> Also note that it seems that this internal API is also used by yt-dlp to download Twitch videos.

Thanks @AudricV, I appreciate the update!

**Comment by AudricV** [comment]
> If I understand you right you're advising against the use of Twire?

Not at all, if Twire devs want to use the official API, that's their decision, which is understandable, especially in terms of respect of Twitch TOS (which should be important when using Twitch accounts I think) and API stability.

**Comment by opusforlife2** [comment]
Ah! I understand now. That PR description was written months before Twire made using accounts mandatory. The way it's written gave the impression that the usage of the internal API was problematic _despite_ not logging in. As in IP bans or something similar.

Thanks for the clarification, @AudricV. Sorry about the confusion, @Eiion.

**Comment by Eiion** [comment]
> > If I understand you right you're advising against the use of Twire?
> 
> Not at all, if Twire devs want to use the official API, that's their decision, which is understandable, especially in terms of respect of Twitch TOS (which should be important when using Twitch accounts I think) and API stability.

Oh, because "if Twitch detects an account using the internal API from a third-party client, they may ban that account" sure sounded like it.
But I assume this was meant about the account utilizing the API i.e. Twire devs (not the average user logging into their Twitch account via Twire).
